packages/textkit/tests/run/flatten.test.js

```diff
@@ -120,4 +120,33 @@ describe('run flatten operator', () => {
     expect(runs[1]).toHaveProperty('end', 15);
     expect(runs[1].attributes).toEqual({});
   });
+
+  test('multi overlap merges all props', () => {
+    const runs = flatten([
+      { start: 0, end: 10, attributes: { strike: true } },
+      { start: 5, end: 15, attributes: { style: 'bold' } },
+      { start: 9, end: 20, attributes: { color: 'red' } },
+    ]);
+
+    expect(runs).toHaveLength(5);
+    expect(runs[0]).toHaveProperty('start', 0);
```

Comment:
Not necessary to change, but I just wanted to comment about [jest matchers](https://jestjs.io/docs/expect#expectarraycontainingarray) as a heads up.  You could do something like:

```js
expect(runs).toEqual(
  expect.arrayContaining(
    expect.objectContaining({
      start: 0,
      end: 5,
      attributes: expect.objectContaining({ strike: true }),
    }),
  ),
);
```